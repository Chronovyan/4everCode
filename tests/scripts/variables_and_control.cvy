// Test of variables and control flow in Chronovyan

// Variable declarations and assignments
var counter = 1;
var max = 5;
var message = "Counting: ";

// Control flow with a while loop
print("Starting loop...");
while (counter <= max) {
    if (counter % 2 == 0) {
        print(message + counter + " (even)");
    } else {
        print(message + counter + " (odd)");
    }
    counter = counter + 1;
}

// Final check
print("Loop completed. Final counter value: " + counter);

// Simple conditional
if (counter > max) {
    print("Counter exceeded maximum value as expected.");
} else {
    print("Error: Counter did not exceed maximum value.");
} 