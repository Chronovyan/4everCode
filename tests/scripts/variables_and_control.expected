Starting loop...
Counting: 1 (odd)
Counting: 2 (even)
Counting: 3 (odd)
Counting: 4 (even)
Counting: 5 (odd)
Loop completed. Final counter value: 6
Counter exceeded maximum value as expected. 